src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs

```diff
@@ -1606,6 +1605,90 @@ public async Task GET_ClientDisconnected_ConnectionAbortRaised()
         }
     }
 
+    [ConditionalFact]
+    [MsQuicSupported]
+    public async Task GET_ClientIdlesOut_ConnectionAbortRaised()
+    {
+        // Arrange
+        var connectionClosedTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+        var connectionStartedTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+        var builder = CreateHostBuilder(
+            context =>
+            {
+                return Task.CompletedTask;
+            },
+            configureKestrel: kestrel =>
+            {
+                kestrel.Listen(IPAddress.Parse("127.0.0.1"), 0, listenOptions =>
+                {
+                    listenOptions.Protocols = HttpProtocols.Http3;
+                    listenOptions.UseHttps(TestResources.GetTestCertificate());
+
+                    IMultiplexedConnectionBuilder multiplexedConnectionBuilder = listenOptions;
+                    multiplexedConnectionBuilder.Use(next =>
+                    {
+                        return context =>
+                        {
+                            connectionStartedTcs.SetResult();
+                            context.ConnectionClosed.Register(() => connectionClosedTcs.SetResult());
+                            return next(context);
+                        };
+                    });
+                });
+            });
+
+        // Error codes of this form are reserved for unknown errors
+        const int closeErrorCode = 0xBC; // 0x1f * 5 + 0x21
+
+        // Set a non-standard error code so we can be sure the option was consumed
+        builder.ConfigureServices(services =>
+        {
+            services.Configure<QuicTransportOptions>(options =>
+            {
+                options.DefaultCloseErrorCode = closeErrorCode;
+            });
+        });
+
+        var logTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+
+        TestSink.MessageLogged += context =>
+        {
+            if (context.LoggerName == "Microsoft.AspNetCore.Server.Kestrel.Transport.Quic" &&
+                context.EventId.Id == 6) // application aborted connection
+            {
+                Assert.Contains($"aborted by application with error code {closeErrorCode}", context.Message);
+                logTcs.SetResult();
+            }
+        };
+
+        using (var host = builder.Build())
+        {
+            await host.StartAsync();
+
+            var client = HttpHelpers.CreateClient(idleTimeout: TimeSpan.FromSeconds(1));
```

Comment:
If you just want to test that `MultiplexedConnectionContext.ConnectionClosed` fires and that there's a log from the Quic transport with the expected error code, you could use `QuicConnection.ConnectAsync` and call `CloseAsync(errorCode)` rather than rely on `QuicConnectionOptions.IdleTimeout` like SocketsHttpHandler does. This is a little less end-to-end since it doesn't test the interaction with HttpClient, but it should help the test run faster and potentially make it more reliable.

I'm a little surprised that `await connectionClosedTcs.Task.DefaultTimeout();` is timing out. The default timeout is 30 seconds, right? Does msquic check for idle connections super infrequently? @nibanks

Reply:
I ended up deleting the test since (a) we can't reliably and performantly test client-idles-out scenarios and (b) we no longer munge the error code.